plugins/modules/artisan.py

```diff
@@ -0,0 +1,322 @@
+            - Path to the PHP executable on the remote host if PHP is not included in PATH.
+        required: false
+        type: path
+
+requirements:
+    - Laravel
+    - PHP
+
+notes:
+    - Always appended in each execution are --no-ansi, --no-interaction and --force if available.
+    - Ansible Artisan is intended primarily, but not exclusively, for production deployments
+
+author:
+    - Ralf Langebrake (@codebarista)
+'''
+
+EXAMPLES = r'''
+# php artisan down --secret="t0pS3cRet" --refresh=60 --retry=60 --no-ansi --no-interaction
+- name: Put the application into maintenance / demo mode
+  community.general.artisan:
+    working_dir: "{{ project_src }}"
+    command: down
+    options:
+      - secret="ToPs3cReT"
+      - refresh=60
+      - retry=60
+
+# php artisan storage:link --no-ansi --no-interaction --force
+- name: Connect the public/storage link to storage/app/public
+  community.general.artisan:
+    working_dir: "{{ project_src }}"
+    command: storage:link
+
+# php artisan app:custom-command --no-ansi --no-interaction maybe/a/path
+- name: Run custom command with arguments
+  community.general.artisan:
+    working_dir: "{{ project_src }}"
+    php_path: /usr/local/bin/php
+    command: app:custom-command
+    arguments: maybe/a/path
+  changed_when: "'modified' in app_custom_command.stdout"
+  register: app_custom_command
+
+# php artisan migrate --database=sqlite --isolated=true --seed --force
+- name: Run migrations and seed database
+  community.general.artisan:
+    working_dir: "{{ project_src }}"
+    command: "{{ item }}"
+    options:
+      - database=sqlite
+      - isolated=true
+  loop:
+    - migrate
+    - seed
+
+# php artisan up --no-ansi --no-interaction
+- name: Clear caches and bring the application out of maintenance mode
+  community.general.artisan:
+    working_dir: "{{ project_src }}"
+    command: "{{ item }}"
+  loop:
+    - cache:clear
+    - event:clear
+    - route:clear
+    - view:clear
+    - up
+'''
+
+import os
+import re
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+def parse(out):
+    # remove any unicode strings
+    pattern = re.compile(r'(?:\x1B[@-_]|[\x80-\x9F])[0-?]*[ -/]*[@-~]')
+
+    # split and trim lines
+    lines = list(filter(None, out.splitlines()))
+    message = stderr = lines[0]
+
+    # second line has the exception message
+    if len(lines) > 1:
+        stderr = lines[1]
+
+    # message, stderr, stderr_lines
+    return [
+        re.sub(pattern, '', message).strip(),
+        re.sub(pattern, '', stderr).strip(),
+        list(map(str.strip, lines))
+    ]
+
+
+def has_changed(message):
+    # message fragments that Laravel gives us
+    artisan_messages = [
+        'No publishable',  # resources for tag
+        'Nothing to',  # migrate,...
+        'already',  # up, down, exists,...
+        'Not modified',
+        'No changes'
+    ]
+
+    for info in artisan_messages:
+        if info in message:
+            return False
+
+    return True
```

Comment:
This logic looks very elementary and like something that can easily detect the wrong result. Especially words like "already" can show up in random english sentences.

Reply:
I will use the specific messages in the future.